src/main/groovy/org/cadixdev/gradle/licenser/LicenseExtension.groovy

```diff
@@ -103,16 +104,39 @@ class LicenseExtension extends LicenseProperties {
     @Incubating
     final NamedDomainObjectContainer<LicenseTaskProperties> tasks
 
-    LicenseExtension(Project project) {
-        super(new PatternSet())
+    @PackageScope final ObjectFactory objects
+    @PackageScope final TextResourceFactory textResources
+    @PackageScope final ProviderFactory providers
 
-        this.tasks = project.container(LicenseTaskProperties) { name ->
-            new LicenseTaskProperties((filter as PatternSet).intersect(), name)
+    @Inject
+    LicenseExtension(final ObjectFactory objects, final Project project) {
+        super(new PatternSet(), objects, project.resources.text)
+
+        this.objects = objects
+        this.textResources = project.resources.text
+        this.providers = project.providers
+        this.tasks = objects.domainObjectContainer(LicenseTaskProperties) { String name ->
+            new LicenseTaskProperties((filter as PatternSet).intersect(), name, objects, textResources, charset)
         }
+        this.conditionalProperties = objects.listProperty(LicenseProperties)
 
         // Defaults
-        newLine = true
-        charset = 'UTF-8'
+        this.keywords = objects.listProperty(String).convention(['Copyright', 'License'])
+        this.ignoreFailures = objects.property(Boolean).convention(false)
+        this.skipExistingHeaders = objects.property(Boolean).convention(false)
+        this.charset = objects.property(String).convention('UTF-8')
+        super.charset.set(this.charset)
+
+        def defaultLineEnding
+        try {
+            // Gradle 6+
+            defaultLineEnding = project.providers.systemProperty("line.separator")
+        } catch (final MissingMethodException ex) {
```

Comment:
With all the effort you've put into this PR, to integrate licenser properly with newer Gradle versions, I think it would be better to raise the requirements once rather than again in the next release. Gradle 6.0 seems already > 1.5 years old - why do we still want to support Gradle 5.x?

Reply:
I've sort of gotten in the habit of not dropping support for things without a good reason -- 4.x was dropped here because it has super limited lazy property API, but 5.x was fairly simple to support still, so I kept support.

Reply:
Personally I think these `try { ... } catch` look rather ugly. But @jamierocks does most of the maintenance now so I leave it up to him.

Reply:
I'm happy to keep them for the duration of 0.6.x. I agree that they're ugly - but I think they're a reasonable compromise so we can maintain 5.x support, and use the newfangled APIs Gradle's introduced in more recent version.